tests/integration/kvm.py

```diff
@@ -185,34 +170,34 @@ def _adjust_kvm(self, ssh_inject):
 
         # Command list for adjustments
         cmd_kvm_adj = []
-        # Only inject SSH keys if they're not
-        # already present in .raw image
-        if ssh_inject:
-            cmd_kvm_adj.append("guestfish -a {image} -i mkdir /root/.ssh".format(
-              image=image))
-            cmd_kvm_adj.append("virt-copy-in -a {image} {authorized_keys_file} /root/.ssh/".format(
-              image=image, authorized_keys_file=authorized_keys_file))
-            cmd_kvm_adj.append("guestfish -a {image} -i chown 0 0 /root/.ssh".format(
-              image=image))
-            cmd_kvm_adj.append("guestfish -a {image} -i chown 0 0 /root/.ssh/authorized_keys".format(
-              image=image))
-            cmd_kvm_adj.append("guestfish -a {image} -i chmod 0700 /root/.ssh".format(
-              image=image))
-            cmd_kvm_adj.append("guestfish -a {image} -i chmod 0600 /root/.ssh/authorized_keys".format(
-              image=image))
+        # Create a snapshot image and inject SSH key
+        cmd_kvm_adj.append("qemu-img create -f qcow2 -F raw -b {image} {image}.snapshot.img".format(
```

Comment:
I would probably store the snapshot to /tmp so that it does not leave the test container. This way, we do not face problems with permissions once we would store it next to the original image.